### PR TITLE
Add ability to start service with string actions

### PR DESCRIPTION
**Pull request by volkansahin45**
* String actions like ACTION_RELOAD_REQUIREMENTS

### CODE REVIEW

File: library/core/src/main/java/com/google/android/exoplayer2/offline/DownloadService.java [hunk 1]
```diff
@@ -195,6 +195,27 @@ public static void startWithAction(
...
+  /**
+   * Starts the service, adding an action to be executed.
+   *
+   * @param context A {@link Context}.
+   * @param clazz The concrete download service to be started.
+   * @param action The action string to be executed.
+   *  Can be one of {@link #ACTION_INIT} {@link #ACTION_ADD} {@link #ACTION_RELOAD_REQUIREMENTS} {@link #ACTION_RESTART}
+   * @param foreground Whether the service is started in the foreground.
+   */
+  public static void startWithAction(Context context,
```
Comment (erdemguven) [inline library/core/src/main/java/com/google/android/exoplayer2/offline/DownloadService.java:207]: Why do you need this method? 
There are already dedicated methods  which use ACTION_INIT and ACTION_ADD. 
ACTION_RESTART is an internal action.
Looks like we forgot to provide a convenient method for ACTION_RELOAD_REQUIREMENTS but it has been already removed in the development branch.

**Review by alikemalocalan — APPROVED** [review]

File: library/core/src/main/java/com/google/android/exoplayer2/offline/DownloadService.java (hunk 1 above)
Comment (volkansahin45) [reply library/core/src/main/java/com/google/android/exoplayer2/offline/DownloadService.java:207]: Why ACTION_RELOAD_REQUIREMENTS removed?

Comment (erdemguven) [reply library/core/src/main/java/com/google/android/exoplayer2/offline/DownloadService.java:207]: There is now setRequirements() method in DownloadManager so requirements can be set directly.

Comment (volkansahin45) [reply library/core/src/main/java/com/google/android/exoplayer2/offline/DownloadService.java:207]: Okey thank you.

**Pull request closed without merging**